docs/source/blogs/release-0-4-0-blog/release-0-4-0-blog.rst

```diff
@@ -0,0 +1,290 @@
+
+`GFF3`_ is a sequence annotation flat file format proposed by the `Sequence Ontology Project <http://www.sequenceontology.org/>`_ in 2013, which since has become the de facto format for genome annotation and is widely used by genome browsers and databases such as NCBI `RefSeq <https://www.ncbi.nlm.nih.gov/refseq/>`_ and `GenBank <https://www.ncbi.nlm.nih.gov/genbank/>`_. `GFF3`_, a 9-column tab-separated text format, typically carries the majority of the annotation data in the ninth column, called ``attributes``, as a semi-colon-separated list of ``<tag>=<value>`` entries. As a result, although GFF3 files can be read as Spark DataFrames using Spark SQL's standard ``csv`` data source, the schema of the resulting DataFrame would be quite unwieldy for query and data manipulation of annotation data, because the whole list of attribute tag-value pairs for each sequence will appear as a single semi-colon-separated string in the ``attributes`` column of the DataFrame.
+
+Glow 0.4 has been equipped with the new and flexible ``gff`` Spark SQL data source to address this challenge and create a smooth GFF3 ingest and query experience. While reading the GFF3  file, the ``gff`` data source parses the ``attributes`` column of the file to create an appropriately typed column for each tag, which, in each row, will contain the value corresponding to that tag in that row (or ``null`` if the tag does not appear in the row). Consequently, all tags in the GFF3 ``attributes`` column will have their own corresponding column in the Spark DataFrame, making annotation data query and manipulation much easier.
+
+.. _gff3_ingest:
+
+Ingesting GFF3 Annotation Data
+~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Like any other Spark data source, reading GFF3 files using Glow's ``gff`` data source can be done in a single line of code. As an example, the sequence annotation data for the Homo Sapiens genome assembly GRCh38.p13 in GFF3 format (obtained from `RefSeq ftp site <https://ftp.ncbi.nlm.nih.gov/genomes/refseq/vertebrate_mammalian/Homo_sapiens/reference/GCF_000001405.39_GRCh38.p13/>`_ ) can be ingested using the ``gff`` reader as follows, resulting in the ``annotations_df`` DataFrame shown in :numref:`fig_annotations_df`.
+
+.. _annotations_df:
+
+.. code-block::
+
+  import glow
+  glow.register(spark)
+
+  gff_path = '/databricks-datasets/genomics/gffs/GCF_000001405.39_GRCh38.p13_genomic.gff.bgz'
+
+  annotations_df = spark.read.format('gff').load(gff_path) \
+      .filter("seqid = 'NC_000022.11'") \
+      .alias('annotations_df')
+
+.. figure:: annotations_df.png
+   :align: center
+   :width: 800
+   :name: fig_annotations_df
+
+   A small section of the ``annotations_df`` DataFrame
+
+Here, we have additionally filtered the results to chromosome 22 and given an alias to the resulting DataFrame to use it later in continuation of our example.
```

Comment:
`Here, we have aliased the DataFrame to "annotations_df" and filtered to annotations on chromosome 22.`